packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -465,6 +606,7 @@ class RenderEditable extends RenderBox {
   void detach() {
     _offset.removeListener(markNeedsPaint);
     _showCursor.removeListener(markNeedsPaint);
+    RawKeyboard.instance.removeListener(_handleKeyEvent);
```

Comment:
this should presumably only happen if we're focused

Reply:
attach also needs to be handled

Reply:
So right now I'm adding and removing based on focus. So this line will only execute if the text field isn't unfocused, but application focus changes. So it's definitely possible this executes even when the listener has already been removed. Which I think is fine?

Reply:
We shouldn't remove listeners that aren't added.

Reply:
So how can I account for the case where the editable is detached, but not unfocused? Or will that always be the case?

Reply:
If it's detached, it can't be focused, because it's not in the tree (and thus not on the screen).

Reply:
"detach" basically is the equivalent of "dispose", except that it's possible for it to be "attach"ed again afterwards. RenderObjects start in the "detached" state.